lib/facter/gluster.rb

```diff
@@ -78,11 +73,12 @@
           end
         end
       end
+      # rubocop:disable Metrics/BlockNesting
       if volume_options
         volume_options.each do |vol, opts|
           Facter.add("gluster_volume_#{vol}_options".to_sym) do
             setcode do
-              opts.join(',')
+              opts.join(';')
```

Comment:
By using `;` instead of `,` as separator for the volume options I tried to fix this error as less invasive as possible. If it's now possible to use sturctured-facts in this module I can adapt the code to do this. In this case I'd split this PR into 2. One that adds the code for using the xml output of gluster for retrieving the facts. And the other one that converts the volume fact from a string to an array. This will make it also easier to give the PRs a better name :wink: